c7n/executor.py

```diff
@@ -50,7 +50,10 @@ def map(self, func, iterable):
             yield func(args)
 
     def submit(self, func, *args, **kw):
-        return MainThreadFuture(func(*args, **kw))
+        try:
```

Comment:
can we make this behavior conditional.. i definitely have use cases where i use the main thread executor to get synchronous exceptions.